From: Takashi Iwai <tiwai@suse.com>
Date: Tue May 7 14:11:19 2013 +0200
Subject: drm/mgag200: Fix framebuffer pitch calculation.
Patch-Mainline: to be upstreamed
Git-commit: bbb37478d48539ca1aafd605aa7fae15da84ec6f
Git-repo: git://people.freedesktop.org/~airlied/linux/
References: bnc #806990
Signed-off-by: Egbert Eich <eich@suse.com>

The framebuffer pitch calculation needs to be done differently for bpp == 24
- check xf86-video-mga for reference.

Signed-off-by: Egbert Eich <eich@suse.de>
---
 drivers/gpu/drm/mgag200/mgag200_mode.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-3.0-SLE11-SP3.orig/drivers/gpu/drm/mgag200/mgag200_mode.c
+++ linux-3.0-SLE11-SP3/drivers/gpu/drm/mgag200/mgag200_mode.c
@@ -891,7 +891,7 @@ static int mga_crtc_mode_set(struct drm_
 
 	pitch = crtc->fb->pitches[0] / (crtc->fb->bits_per_pixel / 8);
 	if (crtc->fb->bits_per_pixel == 24)
-		pitch = pitch >> (4 - bppshift);
+		pitch = (pitch * 3) >> (4 - bppshift);
 	else
 		pitch = pitch >> (4 - bppshift);
 
